### PR TITLE
Research tests

### LINKED ISSUE #247: Dynamic fetching of data in research tests
Fetch the data that we are asserting dynamically in research tests. (E.g. `fetch the meme coin with largest market cap`)

### PR DESCRIPTION
**Pull request by nerfZael**
Closes #247 #238

- Testing only what the user actually sees
- Testing with dynamic instead of hardcoded data

### CODE REVIEW

**Comment by nerfZael** [comment]
/workflows/benchmarks agents/token/send,agents/token/test_swap.py,agents/token/test_swap_and_send.py 3

**Comment by nerfZael** [comment]
/workflows/benchmarks agents/token/send,agents/token/test_swap.py::test_swap_complex_2 5

**Comment by nerfZael** [comment]
/workflows/benchmarks agents/token/send,agents/token/research

**Comment by nerfZael** [comment]
/workflows/benchmarks agents/token/research

File: autotx/autotx_agent.py [hunk 1]
```diff
@@ -17,7 +18,7 @@ def __init__(self) -> None:
             f"{tool.name}: {tool.description}" for tool in self.tools
         ]
 
-    def build_autogen_agent(self, autotx: 'AutoTx', user_proxy: autogen.UserProxyAgent, llm_config: Optional[Dict[str, Any]]) -> autogen.Agent:
+    def build_autogen_agent(self, autotx: 'AutoTx', user_proxy: autogen.UserProxyAgent, llm_config: Optional[Dict[str, Any]], notify_user: Callable[[str, Color | None], None]) -> autogen.Agent:
```
Comment (cbrzn) [inline autotx/autotx_agent.py:21]: nit: instead of passing `notify_user` isnt it possible to use `autotx.notify_user`?

**Review by cbrzn — APPROVED** [review]

**Pull request merged**